lib/msf/core/post/android/priv.rb

```diff
@@ -0,0 +1,35 @@
+# -*- coding: binary -*-
+
+require 'msf/core/post/common'
+require 'msf/core/post/file'
+require 'msf/core/post/unix'
+
+module Msf
+class Post
+module Android
+module Priv
+
+  include Msf::Post::Common
+
+  public
+
+  # Returns whether we are running as root or not.
+  #
+  # @return [Boolean] TrueClass if as root, otherwise FalseClass.
+  def is_root?
+    id = cmd_exec('id')
+    uid = id.scan(/uid=(\d+)(.+)/).flatten.first
+    if uid =~ /^0$/
```

Comment:
There's a `Regexp#===` method that is handy for returning boolean matches: http://ruby-doc.org/core-2.1.1/Regexp.html#method-i-3D-3D-3D

`/^0$/ === uid`